src/Symfony/Component/AssetMapper/AssetMapper.php

```diff
@@ -0,0 +1,235 @@
+            }
+            $assets[] = $asset;
+        }
+
+        return $assets;
+    }
+
+    public function getAssetFromFilesystemPath(string $filesystemPath): ?MappedAsset
+    {
+        $logicalPath = $this->mapperRepository->findLogicalPath($filesystemPath);
+        if (null === $logicalPath) {
+            return null;
+        }
+
+        return $this->getAsset($logicalPath);
+    }
+
+    public function getFilesystemPath(string $logicalName): ?string
+    {
+        return $this->mapperRepository->find($logicalName);
+    }
+
+    public function getPublicPath(string $logicalPath): ?string
+    {
+        $manifestData = $this->loadManifest();
+        if (isset($manifestData[$logicalPath])) {
+            return $manifestData[$logicalPath];
+        }
+
+        $filePath = $this->mapperRepository->find($logicalPath);
+        if (null === $filePath) {
+            return null;
+        }
+
+        // grab the Asset - first look in the cache, as it may only be partially created
+        $asset = $this->assetsCache[$logicalPath] ?? $this->getAsset($logicalPath);
+        [$digest, $isPredigested] = $this->getDigest($asset);
+
+        if ($isPredigested) {
+            return $this->publicPrefix.$logicalPath;
+        }
+
+        return $this->publicPrefix.preg_replace_callback('/\.(\w+)$/', function ($matches) use ($digest) {
+            return "-{$digest}{$matches[0]}";
+        }, $logicalPath);
+    }
+
+    public static function isPathPredigested(string $path): bool
+    {
+        return 1 === preg_match(self::PREDIGESTED_REGEX, $path);
+    }
+
+    public function getPublicAssetsFilesystemPath(): string
+    {
+        return rtrim(rtrim($this->projectRootDir, '/').'/'.$this->publicDirName.$this->publicPrefix, '/');
+    }
+
+    /**
+     * Returns an array of "string digest" and "bool predigested".
+     *
+     * @return array{0: string, 1: bool}
+     */
+    private function getDigest(MappedAsset $asset): array
+    {
+        // check for a pre-digested file
+        if (1 === preg_match(self::PREDIGESTED_REGEX, $asset->getLogicalPath(), $matches)) {
+            return [$matches[1], true];
+        }
+
+        return [
+            hash('xxh128', $this->calculateContent($asset)),
+            false,
+        ];
+    }
+
+    private function getMimeType(string $logicalPath): ?string
+    {
+        $filePath = $this->mapperRepository->find($logicalPath);
+        if (null === $filePath) {
+            return null;
+        }
+
+        $extension = MappedAsset::extractExtension($logicalPath);
+
+        if (isset(self::EXTENSIONS_MAP[$extension])) {
+            return self::EXTENSIONS_MAP[$extension];
+        }
+
+        if (null === $this->mimeTypes) {
+            throw new \LogicException(sprintf('The file extension "%s" from "%s" does not correspond to any known types so the mime type can\'t be determined. Run "composer require symfony/mime-type" to guess the mime type.', $extension, $logicalPath));
```

Comment:
instead, what about allowing one to configure the map of extensions (via constructor I suppose)?
I don't think using content-based mime-type detection is really useful in the context of assets.
Alternatively, we could skip adding this extension mechanism and wait for users to contribute one (or send PRs to add to the const)

Reply:
That mime type dep really bothered me too. I've replaced it with a configurable "extensions" array where you can pass in extra extensions. Then we'll see if there are any real-world cases where this isn't enough. It is a fairly minor feature anyways: it impacts only (A) the content-type returned by the dev controller (if you have a wacky file type, you probably aren't relying on it being downloaded and understood by your browser anyways) and (B) activating the asset compiles (again, if you have something wacky, it's unlikely you'd want any existing compilers to operate on it anyways).